aten/src/ATen/core/TensorImpl.h

```diff
@@ -97,6 +99,18 @@ struct CAFFE2_API TensorImpl : public c10::intrusive_ptr_target {
 
   template <typename T>
   inline T * data() const {
+    CAFFE_ENFORCE_WITH_CALLER(
+        storage_.data() || numel_ == 0,
+        "The tensor is of non-zero shape, but its data is not allocated yet. "
```

Comment:
perhaps there is a better term to use than "non-zero shape" (is [1.0] non-zero shape?).  Maybe  "has a non-zero number of elements (shape: [...])"?

Reply:
going to use "The tensor has a non-zero number of elements (shape: [2, 1])"